GetIntoTeachingApi/Models/TeacherTrainingAdviser/TeacherTrainingAdviserSignUp.cs

```diff
@@ -63,6 +98,22 @@ public TeacherTrainingAdviserSignUp(Candidate candidate)
             PopulateWithCandidate(candidate);
         }
 
+        private static bool CanSubscribe(Candidate candidate)
+        {
+            if (!candidate.HasTeacherTrainingAdviser())
+            {
+                return true;
+            }
+
+            if (candidate.AdviserStatus == null)
+            {
+                return false;
+            }
+
+            return Enum.GetValues(typeof(ResubscribableAdviserStatus))
```

Comment:
Wondering if `Enum.IsDefined` would work here? http://www.blackwasp.co.uk/EnumIsDefined.aspx

Reply:
Yep good shout; I dismissed that earlier because it wouldn't handle the case when the field is `null` but I didn't get that working with `GetValues` either (for some reason it wouldn't let me cast the `int` of the enum to `int?` of the field 🤷‍♂️) so I broke out the `null` check and stayed with `GetValues`; I'll push up the change